src/webgpu/api/operation/vertex_state.spec.ts

```diff
@@ -0,0 +1,348 @@
+    }
+
+    /* prettier-ignore */
+    const vertexArray = new Float32Array([
+      // float4 position
+      -1.0, 1.0, 0.0, 1.0,
+      1.0, -1.0, 0.0, 1.0,
+      1.0, 1.0, 0.0, 1.0,
+      -1.0, -1.0, 0.0, 1.0
+    ]);
+    const vertexBuffer = this.device.createBuffer({
+      size: vertexArray.byteLength,
+      usage: GPUBufferUsage.VERTEX,
+      mappedAtCreation: true,
+    });
+    new Float32Array(vertexBuffer.getMappedRange()).set(vertexArray);
+    vertexBuffer.unmap();
+
+    const encoder = this.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: this.colorAttachment.createView(),
+          loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+          storeOp: 'store',
+        },
+      ],
+    });
+    pass.setPipeline(pipeline);
+    pass.setVertexBuffer(0, vertexBuffer);
+    pass.setIndexBuffer(indexBuffer, indexFormat, indexOffset);
+    pass.drawIndexed(indexCount, 1, 0, 0, 0);
+    pass.endPass();
+    encoder.copyTextureToBuffer(
+      { texture: this.colorAttachment, origin: { x: 0, y: 0, z: 0 } },
+      { buffer: this.result, bytesPerRow: this.bytesPerRow, rowsPerImage: this.rowsPerImage },
+      { width: kWidth, height: kHeight, depth: 1 }
+    );
+    this.device.defaultQueue.submit([encoder.finish()]);
+  }
+
+  CreateExpectedUint8Array(renderShape: RenderShape): Uint8Array {
+    let texelValueBytes = new Uint8Array([0x00, 0x00, 0x00, 0xff]);
+    const arrayBuffer = new Uint8Array(this.byteLength);
+    for (let row = 0; row < kHeight; row++) {
+      for (let col = 0; col < kWidth; col++) {
+        if (renderShape === RenderShape.Square) {
+          texelValueBytes = new Uint8Array([0x00, 0xff, 0x00, 0xff]);
+        } else if (renderShape === RenderShape.BottomLeftTriangle && row > col) {
+          texelValueBytes = new Uint8Array([0x00, 0xff, 0x00, 0xff]);
+        } else if (
+          renderShape === RenderShape.Points &&
+          (row === 0 || row === kHeight - 1) &&
+          (col === 0 || col === kWidth - 1)
+        ) {
+          texelValueBytes = new Uint8Array([0x00, 0xff, 0x00, 0xff]);
+        } else if (
+          renderShape === RenderShape.XShape &&
+          (col === row || col + row === kWidth - 1)
+        ) {
+          texelValueBytes = new Uint8Array([0x00, 0xff, 0x00, 0xff]);
+        } else {
+          texelValueBytes = new Uint8Array([0x00, 0x00, 0x00, 0xff]);
+        }
+
+        const byteOffset = row * this.bytesPerRow + col * texelValueBytes.byteLength;
+        arrayBuffer.set(texelValueBytes, byteOffset);
+      }
+    }
+    return arrayBuffer;
+  }
+}
+
+export const g = makeTestGroup(IndexFormatTest);
+
+// Test indexing draw with index format of uint16. If this is interpreted as uint32, it will have
+// index 1 and 2 be both 0 and render nothing. And the index buffer size - offset must be not less
+// than the size required by triangle list, otherwise it also render nothing.
```

Comment:
This description should be inside `g.test(...).desc( HERE )` instead of as a comment. Then it will show up in /standalone/ as part of the test description.

Same below.